### ISSUE TITLE
Why the text show like cube

### ISSUE THREAD

**Issue by cipher1985** [ocornut/imgui#2045]
### Version/Branch of Dear ImGui:
v1.63

### Back-end file/Renderer/OS:
glfw + opengl2 + imgui

### My Issue/Question:
why the text show like the attachment?
when  I use this code,it's show text like cube:
**ImGui::GetIO().Fonts->AddFontFromFileTTF("msyh.ttc", 60.0f, NULL, ImGui::GetIO().Fonts->GetGlyphRangesChineseFull());**
![20180827103840](https://user-images.githubusercontent.com/30994177/44639317-d7f06780-a9ed-11e8-8983-3133ed7ae248.jpg)
but I use this code,it's show text like normal:
**ImGui::GetIO().Fonts->AddFontFromFileTTF("msyh.ttc", 60.0f, NULL, ImGui::GetIO().Fonts->GetGlyphRangesChineseSimplifiedCommon());**
![20180827113819](https://user-images.githubusercontent.com/30994177/44639320-d888fe00-a9ed-11e8-8fa8-87f71942fb49.jpg)

**Comment by ice1000** [comment]
It's quite important where you called

```cpp
ImGui::GetIO().Fonts->AddFontFromFileTTF("msyh.ttc", 60.0f, NULL, ImGui::GetIO().Fonts->GetGlyphRangesChineseFull());
```

, please clarify (before mainloop? After `NewFrame`? Lots of possibilities).

**Comment by heroboy** [comment]
I think you use too large font size. And `GetGlyphRangesChineseFull` makes too large texture to be created.

**Comment by ice1000** [comment]
@ocornut is it possible to have lazily loaded font atlas? This looks very difficult to me, but I'm not sure.

**Comment by ocornut** [comment]
Sorry I have to close this.
The top of the New Issue Template says:

“
IF YOU ARE HAVING AN ISSUE [...] ADDING FONTS [...], please post on the "Getting Started" Discourse forum:
https://discourse.dearimgui.org/c/getting-started
“

**Comment by ocornut** [comment]
FYI, work on `dynamic_fonts` branch aimed to be merged in 1.92 (#8465) solves this, as glyphs are rasterized and packed on demand when they are used.
